openssl-dynamic/src/main/c/sslcontext.c

```diff
@@ -1943,6 +1960,259 @@ TCN_IMPLEMENT_CALL(void, SSLContext, setCertificateCallback)(TCN_STDARGS, jlong
 #endif // defined(LIBRESSL_VERSION_NUMBER)
 }
 
+// Support for SSL_PRIVATE_KEY_METHOD.
+#ifdef OPENSSL_IS_BORINGSSL
+
+enum ssl_private_key_result_t tcn_private_key_sign_java(SSL *ssl, uint8_t *out, size_t *out_len, size_t max_out, uint16_t signature_algorithm, const uint8_t *in, size_t in_len) {
+    enum ssl_private_key_result_t ret = ssl_private_key_failure;
+
+    tcn_ssl_task_t* sslTask = NULL;
+    tcn_ssl_ctxt_t* c = tcn_SSL_get_app_data2(ssl);
+    jbyteArray resultBytes = NULL;
+    jbyteArray inputArray = NULL;
+    jbyteArray keyArray = NULL;
+    jbyte* b = NULL;
+    int arrayLen = 0;
+    JNIEnv *e = NULL;
+    unsigned char * buf = NULL;
+    int bufLength = 0;
+    EVP_PKEY *key = NULL;
+
+    tcn_get_java_env(&e);
+
+    // key may be NULL.
+    if ((key = SSL_get_privatekey(ssl)) != NULL) {
+        if ((bufLength = i2d_PrivateKey(key, &buf)) <= 0) {
+            goto complete;
```

Comment:
How about making the code below `complete:` as an inline function and using `return` in lieu of `goto`?

Reply:
sorry not sure what you suggest... can you give an example ?

Reply:
This code was actually removed